eox_core/social_tpa_backends.py

```diff
@@ -199,3 +201,83 @@ def get_user_id(self, *args, **kwargs):
                 LOG.info("Updating uid: %s to %s", uid, slug_uid)
 
         return slug_uid
+
+
+class BaseOAuth2PKCEMixin:
+    """
+    TO-DO: Use the `from social_core.backends.oauth import BaseOAuth2PKCE` base class once the pull request is merged: https://github.com/python-social-auth/social-core/pull/856/files#diff-d44db201b48f2ec7cab2a0c981213a2991630567778cc6608d03fa0e3804e466R467
+    Base class for providers using OAuth2 with Proof Key for Code Exchange (PKCE).
+    OAuth2 details at:
+        https://datatracker.ietf.org/doc/html/rfc6749
+    PKCE details at:
+        https://datatracker.ietf.org/doc/html/rfc7636
+    """
+
+    PKCE_DEFAULT_CODE_CHALLENGE_METHOD = "s256"
+    PKCE_DEFAULT_CODE_VERIFIER_LENGTH = 32
+    USE_PKCE = True
```

Comment:
I don't understand why is this necessary, I mean the obvious answer is because you can disable that behavior on  sub-classes, but if I don't want that behavior on sub-classes I could inherit from  BaseOAuth2 instead of this one